lib/rspec-puppet/adapters.rb

```diff
@@ -186,6 +188,12 @@ def manifest
     end
 
     class Adapter4X < Adapter40
+      def setup_puppet(example_group)
+        super
+
+        Object.send(:const_set, :FacterImpl, Facter) unless defined? FacterImpl
```

Comment:
Can you explain the reasoning for this? It's not obvious to me. Is this to create a `FacterImpl` on the global scope?

Reply:
You're right, since now we use `FacterImpl` instead of `Facter` throughout the code. In short, this is needed to support puppet4-5 (which I don't know if we want, but I figured I'd add it anyway).

Reply:
Since the source code still has it, I'm very happy you did. IMHO it should either be supported or removed, not bit rot inside the repo.

I suppose this works, but wonder if we should add a code comment. I can imagine not every developer knows what this does. Perhaps wrap it in a function like `set_global_facter_impl_unless_defined(value)`?